src/gc/gc.cpp

```diff
@@ -4496,7 +4503,8 @@ void* virtual_alloc (size_t size)
         flags = VirtualReserveFlags::WriteWatch;
     }
 #endif // !FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
-    void* prgmem = GCToOSInterface::VirtualReserve (requested_size, card_size * card_word_width, flags);
+
+    void* prgmem = use_large_pages_p ? GCToOSInterface::VirtualCommitLargePages(requested_size, card_size * card_word_width) : GCToOSInterface::VirtualReserve(requested_size, card_size * card_word_width, flags);
```

Comment:
Why do we commit large pages but just resever regular pages?

Reply:
on windows you cannot reserve large pages then commit, you must commit up front. on linux you can but there's a much bigger chance that you will not be able to get a large page later. to make things simple we just commit up front to begin with. can always optimize on linux if desired.

Reply:
Ah, ok, thanks for the explanation.